services/pull/merge.go

```diff
@@ -32,9 +33,89 @@ import (
 	issue_service "code.gitea.io/gitea/services/issue"
 )
 
+// GetDefaultMergeMessage returns default message used when merging pull request
+func GetDefaultMergeMessage(baseGitRepo *git.Repository, pr *models.PullRequest, mergeStyle repo_model.MergeStyle) (string, error) {
+	if err := pr.LoadHeadRepo(); err != nil {
```

Comment:
If Head does not exist then this will not return an error but pr.HeadRepo will be nil